src/application.coffee

```diff
@@ -218,9 +218,15 @@ fetch = (app, setDeviceUpdateState = true) ->
 			Promise.join utils.getConfig('apiKey'), utils.getConfig('uuid'), (apiKey, uuid) ->
 				if conf['RESIN_SUPERVISOR_DELTA'] == '1'
 					logSystemEvent(logTypes.downloadAppDelta, app)
-					requestTimeout = checkInt(conf['RESIN_SUPERVISOR_DELTA_REQUEST_TIMEOUT'], positive: true) ? 30 * 60 * 1000
-					totalTimeout = checkInt(conf['RESIN_SUPERVISOR_DELTA_TOTAL_TIMEOUT'], positive: true) ? 24 * 60 * 60 * 1000
-					dockerUtils.rsyncImageWithProgress(app.imageId, { requestTimeout, totalTimeout, uuid, apiKey }, onProgress)
+					deltaOpts = {
+						uuid, apiKey
+						# use user-defined timeouts, but fallback to defaults if none is provided.
+						requestTimeout: checkInt(conf['RESIN_SUPERVISOR_DELTA_REQUEST_TIMEOUT'], positive: true)
+						applyTimeout: checkInt(conf['RESIN_SUPERVISOR_DELTA_APPLY_TIMEOUT'], positive: true)
```

Comment:
Oh right, I see now. Good catch 👍 

Any preference for a default? Something like 5 minutes (as it was at some point I think?) -- note that the timeout will start counting from the moment the stream to rsync closes, i.e. after download succeeds or fails.